server.go

```diff
@@ -1686,7 +1686,20 @@ func newServer(cfg *Config, listenAddrs []net.Addr,
 // UpdateEstimatorValue is a callback function to change estimator value in the  
 // main cfg when the lncli setmccfg is called to change the estimator value. 
  func (s *server) UpdateEstimatorValue(estimator routing.Estimator) {
-	s.cfg.Estimator = estimator
+	switch c := estimator.Config().(type) {
+	case *routing.AprioriConfig:
+		targetCfg := s.cfg.SubRPCServers.RouterRPC.AprioriConfig
+		targetCfg.PenaltyHalfLife = c.PenaltyHalfLife
+		targetCfg.Weight = c.AprioriWeight
+		targetCfg.CapacityFraction = c.CapacityFraction
+		targetCfg.HopProbability = c.AprioriHopProbability
+
+	case *routing.BimodalConfig:
+		targetCfg := s.cfg.SubRPCServers.RouterRPC.BimodalConfig
+		targetCfg.Scale = int64(c.BimodalScaleMsat)
+		targetCfg.NodeWeight = c.BimodalNodeWeight
+		targetCfg.DecayTime = c.BimodalDecayTime
```

Comment:
I believe this is missing for the updates to be complete:

```go
		targetCfg.ProbabilityEstimatorType =
			routing.AprioriEstimatorName
```

Reply:
There is no ProbabilityEstimatorType on targetCfg!!!

Reply:
Correct. It should be this:

```go
s.cfg.SubRPCServers.RouterRPC.RoutingConfig.ProbabilityEstimatorType =
			routing.AprioriEstimatorName
```

Reply:
Perfect ... it is working now! You were right about the [pointer] (https://github.com/lightningnetwork/lnd/pull/8857#discussion_r1663641513)